x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/job/RollupJobTask.java

```diff
@@ -261,16 +285,21 @@ public synchronized void start(ActionListener<StartRollupJobAction.Response> lis
      * @param listener The listener that is requesting the stop, so that we can signal completion/failure
      */
     public synchronized void stop(ActionListener<StopRollupJobAction.Response> listener) {
+
         final IndexerState newState = indexer.stop();
         switch (newState) {
             case STOPPED:
                 listener.onResponse(new StopRollupJobAction.Response(true));
                 break;
 
             case STOPPING:
-                // update the persistent state only if there is no background job running,
-                // otherwise the state is updated by the indexer when the background job detects the STOPPING state.
-                RollupJobStatus state = new RollupJobStatus(IndexerState.STOPPED, indexer.getPosition());
+                // update the persistent state to STOPPED.  There are two scenarios and both are safe:
+                // 1. we persist STOPPED now, indexer continues a bit then sees the flag and checkpoints another
+                //    STOPPED with the more recent position.  That will also upgrade the ID scheme
+                // 2. we persist STOPPED now, indexer continues a bit but then dies.  When/if we resume we'll pick up
+                //    at last checkpoint, overwrite some docs and eventually checkpoint.  At that time we'll also
+                //    upgrade the ID scheme
+                RollupJobStatus state = new RollupJobStatus(IndexerState.STOPPED, indexer.getPosition(), upgradedDocumentID.get());
```

Comment:
This is just a clarification because the old comment was equal parts vague, confusing and a bit wrong.